pkg/agent/agent.go

```diff
@@ -338,7 +349,19 @@ func (c *agentController) generateExtManagedKubeconfigSecret(ctx context.Context
 	klusterletNamespace := &corev1.Namespace{}
 	klusterletNamespaceNsn := types.NamespacedName{Name: "klusterlet-" + managedClusterAnnoValue}
 
-	err := c.spokeClient.Get(ctx, klusterletNamespaceNsn, klusterletNamespace)
+	//When klusterlet for imported HC is created, namespace might not exist right away, attempt twice before failing
+	var err error
+	for try := 1; try <= 5; try++ {
```

Comment:
This can potentially block the agent to reconcile other requests for 5 minutes. I think we should just return error without retries and let the reconcile re-queue.

Reply:
@rokej Should I set a retry limit so that this can run https://github.com/stolostron/hypershift-addon-operator/blob/e20ffa7a10bca4abff39c60d19f59dd33f616e2f/pkg/agent/agent.go#L550

Reply:
We should just return any error from `c.spokeClient.Get(ctx, klusterletNamespaceNsn, klusterletNamespace)` without a retry. We should return the error here https://github.com/stolostron/hypershift-addon-operator/blob/e20ffa7a10bca4abff39c60d19f59dd33f616e2f/pkg/agent/agent.go#L554 so that it gets requeued here https://github.com/stolostron/hypershift-addon-operator/blob/e20ffa7a10bca4abff39c60d19f59dd33f616e2f/pkg/agent/agent.go#L605. 

Need to think about the metrics data here https://github.com/stolostron/hypershift-addon-operator/blob/e20ffa7a10bca4abff39c60d19f59dd33f616e2f/pkg/agent/agent.go#L604 and https://github.com/stolostron/hypershift-addon-operator/blob/e20ffa7a10bca4abff39c60d19f59dd33f616e2f/pkg/agent/agent.go#L551 to filter out false-alarms where it fails because the namespace does not exist yet.